__tests__/components/TableDisplay.spec.ts

```diff
@@ -0,0 +1,1480 @@
+                    // Act:
+                    // @ts-ignore - this is a protected method
+                    vm.closeModal('targetValue');
+                }
+
+                await vm.$nextTick();
+
+                // Assert:
+                // @ts-ignore - this is a protected method
+                const formValue = vm.modalFormsVisibility.targetValue;
+                const formComponent = wrapper.findComponent(ModalMetadataUpdate).exists();
+
+                // @ts-ignore - this is a protected method
+                expect(vm.targetedMetadataList).toEqual(mockMetadataValues);
+                expect(formValue).toBe(expectedResult);
+                expect(formComponent).toBe(expectedResult);
+            };
+
+            test('open model for metadata value form', async () => {
+                await runBasicMetadataValueModelTests(true);
+            });
+
+            test('close model for metadata value form', async () => {
+                await runBasicMetadataValueModelTests(false);
+            });
+
+            test('open model for update metadata value', async () => {
+                await runBasicUpdateMetadataValueModelTests(true);
+            });
+
+            test('close model for update metadata value', async () => {
+                await runBasicUpdateMetadataValueModelTests(false);
+            });
+        });
+    });
+
+    describe('loadMore', () => {
+        const runBasicSkipLoadMoreTests = async (assetType) => {
+            // Arrange:
+            const wrapper = getTableRowWrapper(
+                {
+                    currentConfirmedPage: { pageNumber: 3, isLastPage: true },
+                },
+                {
+                    assetType,
+                },
+            );
+
+            const vm = wrapper.vm as TableDisplayTs;
+
+            // Act:
+            vm.loadMore();
+
+            // Assert:
+            expect(vm.$store.dispatch).not.toBeCalledWith('namespace/LOAD_NAMESPACES');
+        };
+
+        test('called when asset type is namespace and current page is not last page', () => {
+            // Arrange:
+            const wrapper = getTableRowWrapper(
+                {
+                    currentConfirmedPage: { pageNumber: 1, isLastPage: false },
+                },
+                {
+                    assetType: 'namespace',
+                },
+            );
+
+            const vm = wrapper.vm as TableDisplayTs;
+
+            // Act:
+            vm.loadMore();
+
+            // Assert:
+            expect(vm.$store.dispatch).toBeCalledWith('namespace/LOAD_NAMESPACES', {
+                pageNumber: 2,
+                pageSize: 10,
+            });
+        });
+
+        test('skip when asset type is namespace and current page is last page', () => {
+            runBasicSkipLoadMoreTests('namespace');
+        });
+
+        test('skip when asset type is not namespace', () => {
+            runBasicSkipLoadMoreTests('mosaic');
```

Comment:
should be NOT last page for this test

Reply:
i meant it should be mosaic and NOT last page (helper is always setting last page, which will skip for all types). i didn't mean to rename helper.

Reply:
I see, now I got it 👍🏼

Reply:
I have rewritten a bit, it should look better now